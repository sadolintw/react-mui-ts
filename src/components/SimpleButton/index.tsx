import React from "react"

const SimpleButton: React.FC = () => {
  return <button className="simpleButton">test</button>
}

export default SimpleButton