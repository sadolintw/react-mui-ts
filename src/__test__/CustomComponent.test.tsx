import React from 'react'
import { render, screen } from '@testing-library/react'
import SimpleButton from '../components/SimpleButton'

// const mock = jest.fn();

const component = 'SimpleButton'

describe('SimpleButton tests', () => {
  it(`should call ${SimpleButton} correctly`, () => {
    const { container } = render(<SimpleButton />);
    expect(container).toBeTruthy();
  })

  it('should have class simpleButton', () => {
    const { container } = render(<SimpleButton />);
    const button = container.getElementsByClassName('simpleButton');
    expect(button.length).toBe(1);
  })

  it('should render correctly', () => {
    const { container } = render(<SimpleButton />);
    expect(container).toMatchSnapshot();
  })
})