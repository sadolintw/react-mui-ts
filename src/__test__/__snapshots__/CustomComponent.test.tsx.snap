// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimpleButton tests should render correctly 1`] = `
<div>
  <button
    class="simpleButton"
  >
    test
  </button>
</div>
`;
